src/pages/ResultsPage.js

```diff
@@ -188,6 +146,73 @@ class BaseResultsPage extends BaseCategoriesPage {
             }
         }
     }
+
+    onBackClick(event: SyntheticInputEvent): void {
+        if (this.refs.component.onGoBack) {
+            this.refs.component.onGoBack(event)
+        }
+        if (!event.defaultPrevented) {
+            this.props.history.goBack();
+        }
+    }
+
+    render(): ReactElement {
+        const Component = this
+            .props
+            .location
+            .pathname
+            .match(/map(\/)?$/) ? ResultsMap
+            : ResultsListPage;
```

Comment:
We should still make this not-hinge on a regex. It's super fragile, and will surprise anyone who thinks editing the router is enough.